src/providers/node.rs

```diff
@@ -367,6 +437,59 @@ impl NodeProvider {
 
         all_deps
     }
+
+    pub fn is_nx_monorepo(app: &App) -> bool {
+        app.includes_file("nx.json")
+    }
+
+    pub fn get_nx_app_name(app: &App, env: &Environment) -> Result<Option<String>> {
+        if let Some(app_name) = env.get_config_variable(NX_APP_NAME_ENV_VAR) {
+            return Ok(Some(app_name));
+        }
+
+        if let Ok(nx_json) = app.read_json::<NxJson>("nx.json") {
+            if let Some(default_project) = nx_json.default_project {
+                return Ok(Some(default_project.as_str().unwrap().to_owned()));
+            }
+        }
+
+        return Err(anyhow!("Could not derive nx app to build and run. Please add a default project to your nx config or set NIXPACKS_{}", NX_APP_NAME_ENV_VAR));
```

Comment:
Tip: use the `bail!` macro here as a shorthand for `return Err(anyhow!`

Reply:
Oh neat 😊! done this.